save-sandbox/src/main/kotlin/com/saveourtool/save/sandbox/config/WebSecurityConfig.kt

```diff
@@ -0,0 +1,123 @@
+/**
+ * Configuration beans for security in different profiles
+ */
+
+package com.saveourtool.save.sandbox.config
+
+import com.saveourtool.save.domain.Role
+import com.saveourtool.save.sandbox.security.ConvertingAuthenticationManager
+import com.saveourtool.save.sandbox.security.CustomAuthenticationBasicConverter
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Profile
+import org.springframework.http.HttpStatus
+import org.springframework.security.access.expression.method.DefaultMethodSecurityExpressionHandler
+import org.springframework.security.access.hierarchicalroles.RoleHierarchy
+import org.springframework.security.access.hierarchicalroles.RoleHierarchyImpl
+import org.springframework.security.access.hierarchicalroles.RoleHierarchyUtils
+import org.springframework.security.config.annotation.method.configuration.EnableReactiveMethodSecurity
+import org.springframework.security.config.annotation.web.reactive.EnableWebFluxSecurity
+import org.springframework.security.config.web.server.SecurityWebFiltersOrder
+import org.springframework.security.config.web.server.ServerHttpSecurity
+import org.springframework.security.crypto.factory.PasswordEncoderFactories
+import org.springframework.security.crypto.password.PasswordEncoder
+import org.springframework.security.web.server.SecurityWebFilterChain
+import org.springframework.security.web.server.authentication.AuthenticationWebFilter
+import org.springframework.security.web.server.authentication.HttpStatusServerEntryPoint
+
+import javax.annotation.PostConstruct
+
+@EnableWebFluxSecurity
+@EnableReactiveMethodSecurity
+@Profile("secure")
+@Suppress("MISSING_KDOC_TOP_LEVEL", "MISSING_KDOC_CLASS_ELEMENTS", "MISSING_KDOC_ON_FUNCTION")
+class WebSecurityConfig(
+    private val authenticationManager: ConvertingAuthenticationManager,
+) {
+    @Autowired
+    private lateinit var defaultMethodSecurityExpressionHandler: DefaultMethodSecurityExpressionHandler
```

Comment:
can it be moved to constructor arguments?

Reply:
Moved